mne/morph.py

```diff
@@ -357,78 +443,61 @@ def as_volume(self, stc, fname=None, mri_resolution=False, mri_space=True,
             The image object.
         """
         if format != 'nifti1' and format != 'nifti2':
-            raise ValueError("invalid format specifier %s. Must be 'nifti1' or"
-                             " 'nifti2'" % format)
+            raise ValueError("invalid format %s, must be 'nifti1' or 'nifti1''"
+                             % format)
         if apply_morph:
             stc = self.__call__(stc)  # apply morph if desired
-        return _stc_as_volume(self, stc, fname=fname,
-                              mri_resolution=mri_resolution,
-                              mri_space=mri_space,
-                              format=format)
-
-    def _update_morph_data(self, data=None, kind=None):
-        """Update morph data and kind."""
-        if data is not None:
-            self.params.update(data)
-
-        if kind is not None:
-            self.kind = kind
-
-    def _compute_morph_data(self, verbose=None):
-        """Compute morph data."""
-        self._update_morph_data(_compute_morph_data(self, verbose=verbose))
+        img = _stc_as_volume(self, stc, mri_resolution=mri_resolution,
+                             mri_space=mri_space, format=format)
+        if fname is not None:
+            import nibabel as nib
+            nib.save(img, fname)
+        return img
```

Comment:
you should also add a __repr__ to the SourceMorph class

Reply:
[It already exists](https://github.com/mne-tools/mne-python/pull/5466/files#diff-f47b10faaf7546ef88aeb16b4496abf5R367)